core/src/test/java/org/apache/iceberg/TestFormatVersions.java

```diff
@@ -23,56 +23,100 @@
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import org.junit.jupiter.api.TestTemplate;
 
 public class TestFormatVersions extends TestBase {
   @Parameters(name = "formatVersion = {0}")
   protected static List<Object> parameters() {
-    return Arrays.asList(1);
+    return Arrays.asList(1, 2);
   }
 
   @TestTemplate
-  public void testDefaultFormatVersion() {
-    assertThat(table.ops().current().formatVersion()).isEqualTo(1);
+  public void testFormatVersionUpgrade() {
+    TableOperations ops = table.ops();
+    TableMetadata base = ops.current();
+    int baseTableVersion = base.formatVersion();
```

Comment:
actually there is `formatVersion` which carries the current version that is being set by `parameters()`. That way we don't need to keep track of `base` or `baseTableVersion`

Reply:
Purging baseTableVersion

Reply:
Purging base